src/nvim/charset.c

```diff
@@ -1732,6 +1732,26 @@ char_u* skiptowhite_esc(char_u *p) {
   return p;
 }
 
+/// Get a number from a string and skip over it, signalling overflows
+///
+/// @param[out]  pp  A pointer to a pointer to char_u.
+///                  It will be advanced past the read number.
+/// @param[out]  nr  Number read from the string.
+///
+/// @return 0 on success, -1 on error
+int getdigits_safe(char_u **pp, intmax_t *nr)
+{
+  errno = 0;
+  *nr = strtoimax((char *)(*pp), (char **)pp, 10);
+
+  if ((*nr == INTMAX_MIN || *nr == INTMAX_MAX)
+      && errno == ERANGE) {
+    return -1;
+  }
+
+  return 0;
```

Comment:
The other functions which return `true`/`false` are predicates, as far as I can tell. Using booleans for reporting failures always seemed a little "un-C-ish" to me, but I don't know the neovim codebase well enough. If you think otherwise, I'll change it.